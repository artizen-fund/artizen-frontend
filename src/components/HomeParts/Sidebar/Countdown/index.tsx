import styled from 'styled-components'
import { default as ReactCountdown } from 'react-countdown'
import { Icon } from '@components'
import { breakpoint, palette } from '@theme'
import { rgba } from '@lib'

export interface CountdownProps {
  date: string
}

const Countdown = ({ date }: CountdownProps) => (
  <Wrapper>
    <Icon>countdown</Icon>
    <ReactCountdown {...{ date }} />
  </Wrapper>
)

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
`

export default Countdown
