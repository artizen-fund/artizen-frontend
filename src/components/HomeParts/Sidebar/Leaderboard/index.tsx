import styled from 'styled-components'
import { Table, TableCell } from '@components'

export interface LeaderboardProps {
  leaderboard: Array<{ name: string; amount: number }>
}

const Leaderboard = ({ leaderboard }: LeaderboardProps) => (
  <Table title="Leaderboard">
    {leaderboard.map((benefactor, i) => (
      <TableCell key={`benefactor-${i}`}>
        <div>
          <div>#{i}</div>
          <Name king={i === 0}>{benefactor.name}</Name>
        </div>
        <div>${benefactor.amount}</div>
      </TableCell>
    ))}
  </Table>
)

const Name = styled.div<{ king: boolean }>`
  position: relative;
  ${props =>
    props.king &&
    `
  &:after {
    content: ' ðŸ‘‘';
    position: absolute;
    right: -30px;
    top: -6px;
  }
`}
`

export default Leaderboard
