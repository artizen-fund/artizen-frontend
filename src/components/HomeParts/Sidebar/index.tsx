import styled from 'styled-components'
import { StickyContent, ProgressBar, Button, Table, TableCell } from '@components'
import { breakpoint, palette } from '@theme'
import { rgba } from '@lib'

const Sidebar = () => (
  <Wrapper>
    <h3>
      Join our <strong>May, 2022</strong> donation drive
    </h3>
    <Content>
      <p>$15,250 raised of $25,000 goal</p>
      <ProgressBar>{55}</ProgressBar>
      <Row>
        <CountdownClock />
        <DonationCount>3.2k</DonationCount>
      </Row>
      <Row>
        <Button onClick={() => console.log('donate!')} size="l1">
          Donate Now
        </Button>
        <Button onClick={() => console.log('share!')} size="l1" outline>
          Share
        </Button>
      </Row>
      <Table title="Leaderboard">
        <TableCell>
          <div>#1</div>
          <div>herp derp</div>
          <div>$69,00</div>
        </TableCell>
        <TableCell>
          <div>#2</div>
          <div>herp derp</div>
          <div>$68,00</div>
        </TableCell>
        <TableCell>
          <div>#3</div>
          <div>herp derp</div>
          <div>$67,00</div>
        </TableCell>
      </Table>
      <Table title="Additional Perks">
        <TableCell>Membership in Artizen DAO</TableCell>
        <TableCell>0.01% of $ART tokens</TableCell>
        <TableCell>Collectable Receipt</TableCell>
        <TableCell>Custom Discord Avatar</TableCell>
      </Table>
    </Content>
  </Wrapper>
)

const Wrapper = styled(props => <StickyContent {...props} />)`
  background-color: ${rgba(palette.white)};
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.12);
  @media (prefers-color-scheme: dark) {
    background-color: ${rgba(palette.barracuda)};
    border-bottom: 1px solid rgba(114, 124, 140, 0.12);
    box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.4);
  }
  border-radius: 16px;

  display: none;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    display: block;
    width: 390px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    width: 500px;
  }
`

const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 22px 32px 32px;
  gap: 32px;
  > * {
    width: 100%;
  }
`

const CountdownClock = styled.div``

const DonationCount = styled.div``

const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

export default Sidebar
