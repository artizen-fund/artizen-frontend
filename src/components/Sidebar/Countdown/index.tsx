import { useState, useEffect } from 'react'
import styled from 'styled-components'
import { default as ReactCountdown } from 'react-countdown'
import { Icon } from '@components'
import { isServer } from '@lib'

export interface CountdownProps {
  date: string
}

const Countdown = ({ date }: CountdownProps) => {
  /* note: This component causes obvious hydration errors.
   *       Utilizing useEffect to make sure the SSR and initial render are identical. */

  const [loaded, setLoaded] = useState(false)
  useEffect(() => {
    if (isServer()) return
    setLoaded(true)
  }, [])
  return (
    <Wrapper>
      <Icon>countdown</Icon>
      {loaded && <ReactCountdown {...{ date }} />}
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
`

export default Countdown
