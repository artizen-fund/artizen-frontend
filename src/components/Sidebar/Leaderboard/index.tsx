import styled from 'styled-components'
import { Button, Table, TableCell } from '@components'

export interface LeaderboardProps {
  leaderboard: Array<{ name: string; amount: number }>
}

const Leaderboard = ({ leaderboard }: LeaderboardProps) => {
  const sideItem = (
    <Button onClick={() => console.error('do something')} outline level={2}>
      See All
    </Button>
  )
  return (
    <Table title="Leaderboard" {...{ sideItem }}>
      {leaderboard.map((benefactor, index) => (
        <TableCell key={`benefactor-${index}`}>
          <div>
            <div>#{index}</div>
            <Name king={index === 0}>{benefactor.name}</Name>
          </div>
          <div>${benefactor.amount.toLocaleString()}</div>
        </TableCell>
      ))}
    </Table>
  )
}

const Name = styled.div<{ king: boolean }>`
  position: relative;
  ${props =>
    props.king &&
    `
  &:after {
    content: ' ðŸ‘‘';
    position: absolute;
    right: -30px;
    top: -6px;
  }
`}
`

export default Leaderboard
