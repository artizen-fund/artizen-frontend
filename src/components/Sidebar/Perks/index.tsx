import styled from 'styled-components'
import { Button, Table, TableCell, Glyph } from '@components'
import { rgba } from '@lib'
import { palette, breakpoint } from '@theme'

const Perks = () => {
  const sideItem = (
    <Button href="/" outline level={2}>
      Learn More
    </Button>
  )
  return (
    <Table title="Donor rewards" {...{ sideItem }}>
      <TableCell highlight>
        <div>
          <StyledGlyph glyph="palette" outline />
          Chance to win NFT Artifacts
        </div>
      </TableCell>
      <TableCell highlight>
        <div>
          <StyledGlyph glyph="token" outline />
          Share of $ART tokens
        </div>
      </TableCell>
      <TableCell highlight>
        <div>
          <StyledGlyph glyph="certificate" outline />
          Vote for grant winner
        </div>
      </TableCell>
    </Table>
  )
}

const StyledGlyph = (props: any) => (
  <Wrapper>
    <Glyph {...props} />
  </Wrapper>
)

const Wrapper = styled.div`
  height: 40px;
  width: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: ${rgba(palette.white)};
  @media (prefers-color-scheme: dark) {
    background: ${rgba(palette.slate)};
  }
  border-radius: 9999px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    height: 48px;
    width: 48px;
  }
`

export default Perks
