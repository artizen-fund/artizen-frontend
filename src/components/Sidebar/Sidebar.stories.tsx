import styled from 'styled-components'
import Sidebar from './'

export default {
  title: 'page/home/Sidebar',
  component: Sidebar,
  argTypes: {},
}

const Wrapper = styled.div`
  @media only screen and (max-width: 1199px) {
    &:before {
      content: 'note: the Sidebar is not visible on screens under 1200px';
    }
  }
`

export const SidebarComponent = (props: any) => {
  return (
    <Wrapper>
      <Sidebar {...props} />
    </Wrapper>
  )
}
