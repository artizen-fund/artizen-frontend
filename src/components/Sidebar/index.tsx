import styled from 'styled-components'
import Leaderboard from './Leaderboard'
import Perks from './Perks'
import Countdown from './Countdown'
import { Glyph, ProgressBar, Button, StickyContent, StickyCanvas } from '@components'
import { breakpoint, palette, typography } from '@theme'
import { rgba } from '@lib'

type ISidebar = Pick<ISidebarDonatorsQuery, 'Donations'>

const Sidebar = ({ Donations }: ISidebar) => {
  // note: this is just some placeholder nonsense, not final var names
  const FUND_COUNT = 3.2
  const FUND_AMOUNT = 15250
  const FUND_GOAL = 25000
  const FUND_DATE = 'May, 2022'
  const FUND_DEADLINE = '2022-06-30T00:00:00'

  return (
    <StyledStickyCanvas>
      <Wrapper>
        <Header>
          Join our <strong>{FUND_DATE}</strong> donation drive
        </Header>
        <Content>
          <FundBlock>
            <AmountRaised>
              <span>${FUND_AMOUNT.toLocaleString()}</span> raised of ${FUND_GOAL.toLocaleString()} goal
            </AmountRaised>
            <ProgressBar>{FUND_AMOUNT / FUND_GOAL}</ProgressBar>
            <Row>
              <Countdown date={FUND_DEADLINE} />
              <DonationCount>
                <Glyph glyph="trend" /> <span>{FUND_COUNT}k donations</span>
              </DonationCount>
            </Row>
          </FundBlock>
          <Row>
            <Button onClick={() => alert('todo: implement this')} level={1} stretch glyph="donate">
              Donate
            </Button>
            <Button onClick={() => alert('todo: implement this')} level={1} stretch outline>
              Share Now
            </Button>
          </Row>
          <LargeScreensOnly>
            <Leaderboard {...{ Donations }} />
            <Perks />
          </LargeScreensOnly>
        </Content>
      </Wrapper>
    </StyledStickyCanvas>
  )
}

const StyledStickyCanvas = styled(props => <StickyCanvas {...props} />)`
  grid-area: sidebar;
`

const Wrapper = styled(props => <StickyContent {...props} />)`
  background-color: ${rgba(palette.white)};
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.12);
  @media (prefers-color-scheme: dark) {
    background-color: ${rgba(palette.slate)};
    border-bottom: 1px solid rgba(114, 124, 140, 0.12);
    box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.4);
  }
  border-radius: 0px 0px 16px 16px;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    max-width: 388px;
    top: 92px;
    border-radius: 16px 16px 16px 16px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    max-width: 498px;
    top: 108px;
  }
`

const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 22px 32px 32px 32px;
  gap: 32px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    padding: 32px 40px 40px 40px;
    gap: 40px;
  }
  > * {
    width: 100%;
  }
`

const FundBlock = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 16px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    gap: 24px;
  }
  > * {
    width: 100%;
  }
`

const AmountRaised = styled.div`
  ${typography.label.l1}
  color: ${rgba(palette.barracuda)};
  span {
    ${typography.title.l3}
    color: ${rgba(palette.night)};
    @media (prefers-color-scheme: dark) {
      color: ${rgba(palette.moon)};
    }
  }
`

const DonationCount = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
`

const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  gap: 12px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    gap: 16px;
  }
`

const Header = styled.h3`
  padding: 19px 32px 0px 32px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    padding: 26px 40px 0px 40px;
  }
  ${typography.label.l1}
  font-weight: 500;
  strong {
    font-weight: 600;
  }
`

const LargeScreensOnly = styled.div`
  display: none;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    display: contents;
  }
`

export default Sidebar
