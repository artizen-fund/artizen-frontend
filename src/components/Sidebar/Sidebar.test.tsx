import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Sidebar, { ISidebar } from './'

describe('Sidebar', () => {
  const props: ISidebar = {
    FUND_COUNT: 3.2,
    FUND_AMOUNT: 15250,
    FUND_GOAL: 25000,
    FUND_DATE: 'May, 2022',
    FUND_DEADLINE: '2022-06-30T00:00:00',
    donations: [],
  }
  it('renders Sidebar unchanged', () => {
    const { container } = render(<Sidebar {...props} />)
    expect(container).toMatchSnapshot()
  })
})
