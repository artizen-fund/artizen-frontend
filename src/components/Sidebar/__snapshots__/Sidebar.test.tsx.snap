// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders Sidebar unchanged 1`] = `
<div>
  <section
    class="sc-f67c5638-1 OMake sc-522104b9-0 jRWqmK"
  >
    <div
      class="sc-f67c5638-0 cRlpgy sc-522104b9-1 bftMYt"
    >
      <h3
        class="sc-522104b9-7 kATPcK"
      >
        Join our
         
        <strong>
          , 
        </strong>
         
        donation drive
      </h3>
      <div
        class="sc-522104b9-2 flIpuM"
      >
        <div
          class="sc-522104b9-3 fCAqCx"
        >
          <div
            class="sc-2e76876c-0 eZquMO"
          >
            0
          </div>
          <div
            class="sc-522104b9-6 euksCv"
          >
            <div
              class="sc-522104b9-5 eRngVS"
            >
              <div
                class="sc-149d3190-0 GzuI"
              />
               
              <span>
                0
                 donations
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-522104b9-6 euksCv"
        >
          <button
            class="sc-2a588df4-2 jVJhXF undefined "
          >
            <div
              class="sc-149d3190-0 ewzPBY sc-2a588df4-0 gSwEcZ"
            />
            <span>
              Donate
            </span>
          </button>
          <button
            class="sc-2a588df4-2 ekFgDk undefined "
          >
            <span>
              Share Now
            </span>
          </button>
        </div>
        <div
          class="sc-522104b9-8 kgagiP"
        >
          <div
            class="sc-ecff4829-0 fMTZxv"
          >
            <div
              class="sc-c6d4a71-0 bzRXeC"
            >
              <div
                class="sc-c6d4a71-1 hmYVNc"
              >
                Leaderboard
              </div>
              <button
                class="sc-2a588df4-2 dorpEK undefined "
              >
                <span>
                  See All
                </span>
              </button>
            </div>
          </div>
          <div
            class="sc-ecff4829-0 fMTZxv"
          >
            <div
              class="sc-c6d4a71-0 bzRXeC"
            >
              <div
                class="sc-c6d4a71-1 hmYVNc"
              >
                Additional Perks
              </div>
              <a
                class="sc-2a588df4-1 kFiSF undefined "
              >
                <span>
                  Learn More
                </span>
              </a>
            </div>
            <div
              class="sc-9ed997bc-0 dahNPL"
            >
              <div>
                <div
                  class="sc-b48a7ce9-0 QxMPA"
                >
                  <div
                    class="sc-149d3190-0 dSKnsB"
                  />
                </div>
                Membership in Artizen DAO
              </div>
            </div>
            <div
              class="sc-9ed997bc-0 dahNPL"
            >
              <div>
                <div
                  class="sc-b48a7ce9-0 QxMPA"
                >
                  <div
                    class="sc-149d3190-0 hEvUin"
                  />
                </div>
                0.01% of $ART tokens
              </div>
            </div>
            <div
              class="sc-9ed997bc-0 dahNPL"
            >
              <div>
                <div
                  class="sc-b48a7ce9-0 QxMPA"
                >
                  <div
                    class="sc-149d3190-0 jQAqER"
                  />
                </div>
                Collectable Receipt
              </div>
            </div>
            <div
              class="sc-9ed997bc-0 dahNPL"
            >
              <div>
                <div
                  class="sc-b48a7ce9-0 QxMPA"
                >
                  <div
                    class="sc-149d3190-0 kmdMV"
                  />
                </div>
                Custom Discord Avatar
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
